python/paddle/fluid/tests/unittests/test_detach.py

```diff
@@ -0,0 +1,162 @@
+                initializer=fluid.initializer.Constant(7.0))
+            fc1_b_param_attrs = fluid.ParamAttr(
+                initializer=fluid.initializer.Constant(8.0))
+            fc1 = FC("fc",
+                     1,
+                     num_flatten_dims=1,
+                     param_attr=fc1_w_param_attrs,
+                     bias_attr=fc1_b_param_attrs)
+            data = to_variable(data)
+            x = fc(data)
+            x1 = fc1(x)
+            loss = x1
+            # print(loss, loss.shape)
+            loss.backward()
+            return x.gradient()
+
+    def detach_multi(self):
+        data = self.generate_Data()
+        with fluid.dygraph.guard():
+            fc_w_param_attrs = fluid.ParamAttr(
+                initializer=fluid.initializer.Constant(5.0))
+            fc_b_param_attrs = fluid.ParamAttr(
+                initializer=fluid.initializer.Constant(6.0))
+            fc = FC("fc",
+                    10,
+                    num_flatten_dims=1,
+                    param_attr=fc_w_param_attrs,
+                    bias_attr=fc_b_param_attrs)
+            fc1_w_param_attrs = fluid.ParamAttr(
+                initializer=fluid.initializer.Constant(7.0))
+            fc1_b_param_attrs = fluid.ParamAttr(
+                initializer=fluid.initializer.Constant(8.0))
+            fc1 = FC("fc",
+                     1,
+                     num_flatten_dims=1,
+                     param_attr=fc1_w_param_attrs,
+                     bias_attr=fc1_b_param_attrs)
+            fc2_w_param_attrs = fluid.ParamAttr(
+                initializer=fluid.initializer.Constant(9.0))
+            fc2_b_param_attrs = fluid.ParamAttr(
+                initializer=fluid.initializer.Constant(10.0))
+            fc2 = FC("fc",
+                     1,
+                     num_flatten_dims=1,
+                     param_attr=fc2_w_param_attrs,
+                     bias_attr=fc2_b_param_attrs)
+            data = to_variable(data)
+            x = fc(data)
+            x_detach = x.detach()
+            x1 = fc1(x)
+            x2 = fc2(x_detach)
+            loss = x1 + x2
+            # print(loss, loss.shape)
+            loss.backward()
+            return x.gradient()
+
+    def test_NoDetachMulti_DetachMulti(self):
+        array_no_detach_multi = self.no_detach_multi()
+        array_detach_multi = self.detach_multi()
+
+        assert not np.array_equal(array_no_detach_multi, array_detach_multi)
+
+    def test_NoDetachSingle_DetachMulti(self):
+        array_no_detach_single = self.no_detach_single()
+        array_detach_multi = self.detach_multi()
+        assert np.array_equal(array_no_detach_single, array_detach_multi)
+
+    def test_detach_exception(self):
+        x = fluid.layers.data(name="a", shape=[3, 4], dtype='float32')
+        y = fluid.layers.fc(input=x, size=10, bias_attr=True)
+        try:
+            y_detach = y.detach()
+        except Exception as e:
+            print(e)
```

Comment:
This print seems has no effect for ut, better option is raise an specific exception and catch it here, or do some string match to make sure it raises a predictable exception

Reply:
done